src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -34,8 +35,12 @@ public Task Render(HttpContext context)
     private async Task RenderComponentCore(HttpContext context)
     {
         context.Response.ContentType = RazorComponentResultExecutor.DefaultContentType;
-        _renderer.InitializeStreamingRenderingFraming(context);
-        EndpointHtmlRenderer.MarkAsAllowingEnhancedNavigation(context);
+        var isErrorHandler = context.Features.Get<IExceptionHandlerFeature>() is not null;
+        _renderer.InitializeStreamingRenderingFraming(context, isErrorHandler);
+        if (!isErrorHandler)
+        {
+            EndpointHtmlRenderer.MarkAsAllowingEnhancedNavigation(context);
```

Comment:
Why are we not allowing enhanced navigation in this case? I think we should be allowing it.

Reply:
Ok, I'll remove that part, my goal was to disable as many things as possible that could get in the way.